python-avd/pyavd/_eos_designs/ip_addressing/__init__.py

```diff
@@ -139,8 +149,13 @@ def mlag_l3_ip_primary(self) -> str:
         """
         Return IP for L3 Peerings for MLAG Primary
 
+        If "mlag_peer_l3_ipv4_address" is set, it is used.
         Default pool is "mlag_peer_l3_ipv4_pool"
         """
+
+        if self._mlag_peer_l3_ipv4_address:
+            return self._mlag_peer_l3_ipv4_address
```

Comment:
so this is not 100% correct - the problem here is that you will end up with the same value for primary and secondary which should not be the case

Reply:
Yes, thanks for the hint. Maybe it would be better to set the IPs on the group level like this:

```yml
# If two nodes (and only two) are in the same node_group, they will automatically form an MLAG pair
  node_groups:
    # Definition of a node group that will include two devices in MLAG.
    # Definitions under the group will be inherited by both nodes in the group
    - group: DC1_L3_LEAF1
      # ASN to be used by BGP for the group. Both devices in the MLAG pair will use the same BGP ASN
      bgp_as: 65101
      mlag_peer_primary_ipv4_address: 192.168.42.40
      mlag_peer_secondary_ipv4_address: 192.168.42.41
      mlag_peer_l3_primary_ipv4_address: 192.168.43.40
      mlag_peer_l3_secondary_ipv4_address: 192.168.43.41
      nodes:
        # Definition of hostnames under the node_group
        - name: dc1-leaf1a
          id: 1
          mgmt_ip: 172.16.1.101/24
          loopback_ipv4_address: 172.16.1.101/24
          vtep_loopback_ipv4_address: 172.16.1.101/24
          # Definition of the port to be used in the uplink device facing this device.
          # Note that the number of elements in this list must match the length of 'uplink_switches' as well as 'uplink_interfaces'
          uplink_switch_interfaces:
            - Ethernet1
            - Ethernet1
        - name: dc1-leaf1b
          id: 2
          mgmt_ip: 172.16.1.102/24
          uplink_switch_interfaces:
            - Ethernet2
            - Ethernet2
```
An alternative might be something like this, but I think the "group" level would be better. What do you think?

```python
def mlag_ip_primary(self) -> str:
  if self.shared_utils.mlag_role == "primary":
              if self._mlag_peer_ipv4_address:
                  return self._mlag_peer_ipv4_address
          else:
              if self.shared_utils.get_mlag_peer_fact("mlag_peer_ipv4_address"):
                  return self.shared_utils.get_mlag_peer_fact("mlag_peer_ipv4_address")
```

Reply:
IMO we should just have the direct ip options for those who wants to control things. They have to set it per node or build inline jinja.
Note we already have the option of reusing the same mlag subnet under `fabric_ip_addressing`, and this could be used together with setting `mlag_peer_ipv4_pool` per node group to override things per group.